src/Polly.Shared/Duration/DecorrelatedJitterBackoff.cs

```diff
@@ -0,0 +1,118 @@
+
+        /// <summary>
+        /// Whether the first retry will be immediate or not.
+        /// </summary>
+        public bool FastFirst { get; }
+
+        /// <summary>
+        /// The minimum duration value to use for each retry.
+        /// </summary>
+        public TimeSpan MinDelay { get; }
+
+        /// <summary>
+        /// The maximum duration value to use for each retry.
+        /// </summary>
+        public TimeSpan MaxDelay { get; }
+
+        /// <summary>
+        /// Creates a new instance of the class.
+        /// </summary>
+        /// <param name="minDelay">The minimum duration value to use for each retry.</param>
+        /// <param name="maxDelay">The maximum duration value to use for each retry.</param>
+        /// <param name="fastFirst">Whether the first retry will be immediate or not.</param>
+        /// <param name="random">An optional <see cref="Random"/> instance to use.
+        /// If not specified, will use a shared (singleton) instance with a random seed.</param>
+        public DecorrelatedJitterBackoff(TimeSpan minDelay, TimeSpan maxDelay, bool fastFirst = false, Random random = null)
+        {
+            if (minDelay < TimeSpan.Zero) throw new ArgumentOutOfRangeException(nameof(minDelay));
+            if (maxDelay < minDelay) throw new ArgumentOutOfRangeException(nameof(maxDelay));
+
+            _random = random ?? s_random;
+
+            MinDelay = minDelay;
+            MaxDelay = maxDelay;
+            FastFirst = fastFirst;
+        }
+
+        /// <summary>
+        /// Generate the sequence of <see cref="TimeSpan"/> values to use as sleep-durations.
+        /// </summary>
+        /// <param name="retryCount">The maximum number of retries to use, in addition to the original call.</param>
+        public IReadOnlyList<TimeSpan> Discrete(int retryCount)
+        {
+            if (retryCount < 0) throw new ArgumentOutOfRangeException(nameof(retryCount));
+
+            TimeSpan[] delays = new TimeSpan[retryCount];
+            if (delays.Length == 0)
+                return delays;
+
+            int i = 0;
+            if (FastFirst)
+                delays[i++] = TimeSpan.Zero;
+
+            double range = MaxDelay.TotalMilliseconds - MinDelay.Milliseconds;
+
+            for (; i < delays.Length; i++)
+            {
+                double ms = range * _random.NextDouble(); // Range
+                ms += MinDelay.TotalMilliseconds; // Floor
+                ms = Math.Min(ms, MaxDelay.TotalMilliseconds); // Ceiling
+
+                delays[i] = TimeSpan.FromMilliseconds(ms);
+            }
+
+            return delays;
+        }
+
+        /// <summary>
+        /// Generate a continuous sequence of <see cref="TimeSpan"/> values to use as sleep-durations.
+        /// </summary>
+        /// <param name="retryCount">The maximum number of retries to use, in addition to the original call.</param>
+        public IEnumerable<TimeSpan> Continuous(int retryCount)
```

Comment:
This method is meant to be used in the `Forever()` methods. But not sure if it's adequate or suitable.